src/option.rs

```diff
@@ -95,51 +96,64 @@ impl<T: Miniconf> Miniconf for Option<T> {
         self.0
             .as_ref()
             .map(|value| value.next_path(state, path))
-            .unwrap_or(false)
+            .unwrap_or_default()
     }
 }
 
 impl<T: crate::Serialize + crate::DeserializeOwned> Miniconf for core::option::Option<T> {
-    fn set_path(
+    fn set_path<'a, P: Peekable<Item = &'a str>>(
         &mut self,
-        mut path_parts: core::iter::Peekable<core::str::Split<char>>,
+        path_parts: &mut P,
         value: &[u8],
-    ) -> Result<(), Error> {
+    ) -> Result<usize, Error> {
+        if self.is_none() {
+            return Err(Error::PathNotFound);
+        }
+
         if path_parts.peek().is_some() {
             return Err(Error::PathTooLong);
         }
 
-        *self = serde_json_core::from_slice(value)?.0;
-        Ok(())
+        let (value, len) = serde_json_core::from_slice(value)?;
+        *self = Some(value);
```

Comment:
```suggestion
        *self = value;
```

The deserialized `value` is an `Option` type, and thus may be `null`, in which case it is `None`.

Reply:
For `Option` we only support setting `Some`. Otherwise the user (through Miniconf) could set a `Some` to a `None` exactly once and then never `get()` or `set()` it.

Reply:
This is also a bit of a problem in `serde_json{,_core}`. For `Option<()>` both variants are JSON `null`.